### ISSUE TITLE
Change query support requirement from MUST to SHOULD for properties with support level SHOULD

### ISSUE THREAD

**Issue by rartino** [Materials-Consortia/OPTIMADE#319]
From a call discussion:

We have a number of properties where the query support is marked as MUST, whereas support for the property itself is marked as SHOULD.

The consensus on the call was that this is a specification bug that should be fixed in the next minor release by lowering the requirement level on queries on those properties to SHOULD.

**Comment by merkys** [comment]
I am trying to remember/understand what we meant by using such combination of requirement levels (#210 seems to introduce them). Having OPTIONAL property support with REQUIRED (=MUST) query support makes certain sense:

* If an implementation supports a property, then it MUST support queries on it;
* If an implementation does not support a property, then it MUST treat it as having unknown values and MUST support filtering on that as described in section "Filtering on Properties with an unknown value".

Take `nsites` for example. Specification says:
> **Support**: SHOULD be supported by all implementations, i.e., SHOULD NOT be :val:`null`.
> **Query**: MUST be a queryable property with support for all mandatory filter features.

Thus if we (i.e., the COD) want to include non-`null` values for `nsites`, we also MUST make them queryable using numeric and other filters. If we cannot (and we do not right now), we MUST NOT include non-`null` values for `nsites`. While this paradoxically discourages a provider to include a valuable piece of information, it ensures client developers that if an implementation returns a non-`null` value for such property, this implementation will support meaningful queries for the property in question.

**Comment by rartino** [comment]
@merkys I think what you write could be a rationale for support at SHOULD level + query at MUST level. But are we not in agreement that not preventing implementations from including useful data in responses outweighs the usefulness that you describe for clients? SHOULD anyway means we strongly encourage inclusion. When (if) we implement 'feature sets' #91, I would suggest the first level raises all these SHOULDs to MUSTs.

**Comment by merkys** [comment]
> @merkys I think what you write could be a rationale for support at SHOULD level + query at MUST level. But are we not in agreement that not preventing implementations from including useful data in responses outweighs the usefulness that you describe for clients?

I completely agree with you here. I am just trying to understand the preconditions and severity of this issue.
